### ISSUE TITLE
MA0040 and MA0032 don't highlight static methods with missing cancelation tokens even if there is an override that accepts it

### ISSUE THREAD

**Issue by JakubHac** [meziantou/Meziantou.Analyzer#770]
### Version of the Meziantou.Analyzer NuGet package

2.0.179

### Rule Identifier

MA0040, MA0032 

### Target Framework

netstandard2.1

### C# Language version

C# 9

### Description

MA0040 and MA0032 don't highlight static methods with missing cancelation tokens even if there is an override that accepts it

### Reproduction Steps

Issue is shown in repo
[https://github.com/JakubHac/Meziantou.Analyzer.MS0032.issue](https://github.com/JakubHac/Meziantou.Analyzer.MS0032.issue)

**Comment by meziantou** [comment]
Currently, it does report when there is an overload with exactly one more parameter of type `CancellationToken`. This is not the case of `UniTask.NextFrame` as the other overload has 2 parameters, one being optional.

Repro code:

````c#
Sample.Repro(); // no report as the other overload has 2 more parameters

class Sample
{
    public static void Repro() => throw null;
    public static void Repro(CancellationToken cancellationToken, bool dummy = false) => throw null;
}
````

**Comment by meziantou** [comment]
The latest version should handle this case when you add the following configuration in the editorconfig file

````
MA0032.allowOverloadsWithOptionalParameters = true
````
